presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -592,10 +594,10 @@ private static String keyFromPath(Path path)
     {
         checkArgument(path.isAbsolute(), "Path is not absolute: %s", path);
         String key = nullToEmpty(path.toUri().getPath());
-        if (key.startsWith("/")) {
+        if (key.startsWith(PATH_SEPARATOR)) {
             key = key.substring(1);
```

Comment:
These need to be updated to use the constant instead of assuming the length is `1`